library/src/scala/quoted/Quotes.scala

```diff
@@ -699,10 +699,20 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
         /** TypeRepr of this term */
         def tpe: TypeRepr
 
-        /** Replace Inlined nodes and InlineProxy references to underlying arguments */
+        /** Replace Inlined nodes and InlineProxy references to underlying arguments.
+         *  The resulting tree is useful for inspection of the value or content of a non-inline argument.
+         *
+         *  Warning: This tree may contain references that are out of scope and should not be used in the generated code.
+         *           This method should only used to port Scala 2 that used to access their outer scope unsoundly.
```

Comment:
But it's safe to use this method for macros that don't splice the underlying argument anywhere, right?
Also, it'd be helpful to have an example showing what people should write instead of using underlying/underlyingArgument, because it's not clear to me.

Reply:
> Also, it'd be helpful to have an example showing what people should write instead of using underlying/underlyingArgument, because it's not clear to me.

That is basically all the text in the macro tutorial related to inline arguments, value extraction and quote pattern matching.